agent/action/update_settings_test.go

```diff
@@ -39,35 +40,33 @@ var _ = Describe("UpdateSettings", func() {
 		fileSystem = fakesys.NewFakeFileSystem()
 		platform.GetFsReturns(fileSystem)
 
-		action = NewUpdateSettings(settingsService, platform, certManager, log)
+		action = NewUpdateSettings(settingsService, platform, certManager, log, &agentKiller)
 		newUpdateSettings = boshsettings.UpdateSettings{}
 	})
 
 	AssertActionIsAsynchronous(action)
-	AssertActionIsNotPersistent(action)
+	AssertActionIsPersistent(action)
 	AssertActionIsLoggable(action)
 
-	AssertActionIsNotResumable(action)
+	AssertActionIsResumable(action)
 	AssertActionIsNotCancelable(action)
 
 	Context("on success", func() {
-		It("returns 'updated'", func() {
+		It("returns 'ok'", func() {
 			result, err := action.Run(newUpdateSettings)
 			Expect(err).ToNot(HaveOccurred())
-			Expect(result).To(Equal("updated"))
+			Expect(result).To(Equal("ok"))
 		})
 
 		It("writes the updated settings to a file", func() {
 			action.Run(newUpdateSettings)
-			expectedPath := filepath.Join(platform.GetDirProvider().BoshDir(), "update_settings.json")
-			exists := platform.GetFs().FileExists(expectedPath)
-			Expect(exists).To(Equal(true))
+			Expect(settingsService.SaveUpdateSettingsCallCount).To(Equal(1))
 		})
 	})
 
-	Context("when it cannot write the update settings file", func() {
+	Context("when it save update settings", func() {
```

Comment:
The test setup is basically the same: a situation where writing the settings fails.
Why should we change the test context description here?

Reply:
Looks like a bit of a typo, probably should be something like "when it cannot save update settings".

I'm guessing the change was because the UpdateSettings Action used to be responsible for writing the file to disk itself, but now that responsibility is handled by the settingsService. So these tests are now concerned with the settingsService returning an error (any error), and less concerned with the specifics of a particular error when writing a file.